pkg/zsio/detector/reader.go

```diff
@@ -0,0 +1,43 @@
+package detector
+
+import (
+	"errors"
+	"io"
+
+	"github.com/mccanne/zq/pkg/zsio/ndjson"
+	"github.com/mccanne/zq/pkg/zsio/raw"
+	"github.com/mccanne/zq/pkg/zsio/zjson"
+	zsonio "github.com/mccanne/zq/pkg/zsio/zson"
+	"github.com/mccanne/zq/pkg/zson"
+	"github.com/mccanne/zq/pkg/zson/resolver"
+)
+
+var ErrUnknown = errors.New("input format not recognized")
+
+func NewReader(r io.Reader, n int, t *resolver.Table) (zson.Reader, error) {
```

Comment:
`n` is unused (and so `DetectorMaxRecordSize` can go too i suppose?)

Reply:
yes, I pushed this change while you were reviewing